internal/services/catalog_service.go

```diff
@@ -275,8 +285,34 @@ func (s *CatalogService) CreateBinding(bindingName, org, serviceName string) (in
 		return nil, err
 	}
 
-	return dynamicClient.Resource(serviceBindingGVR).Namespace(deployments.WorkloadsDeploymentID).
+	serviceBinding, err := dynamicClient.Resource(serviceBindingGVR).Namespace(deployments.WorkloadsDeploymentID).
 		Create(context.Background(), obj, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Update the binding secret with kubernetes app labels
+	secret, err := s.GetBindingSecret(bindingName)
```

Comment:
This code seems to assume that the secret is immediately available after the binding resource is created.
I do not believe that we can rely on that. After creating the binding resource
  - It has to be seen by the catalog operator
  - The operator has to talk to the responsible broker (i.e. request and get binding info)
  - And only after all of the above can the operator create the secret with the information it got from the broker.

I would extend the line 295ff to check if the error is `Not Found` and then loop until the secret is actually present
(Maybe a limit of 2 minutes ?) Or maybe place this loop in `GetBindingSecret`.

Reply:
Retracting, now that I saw https://github.com/SUSE/carrier/blob/a07454c02a3705ba74f10be936b9b5d1f5dd6194/internal/services/catalog_service.go#L320

IOW, `GetBindingSecret` already wait.